plutus-core/untyped-plutus-core/src/UntypedPlutusCore/Core.hs

```diff
@@ -1,7 +1,27 @@
-module UntypedPlutusCore.Core
-    ( module Export
-    ) where
+{-# LANGUAGE LambdaCase #-}
+
+module UntypedPlutusCore.Core (
+  module Export,
+  splitApplication,
+  mkApplication,
+) where
 
 import UntypedPlutusCore.Core.Instance as Export
 import UntypedPlutusCore.Core.Plated as Export
 import UntypedPlutusCore.Core.Type as Export
+
+import Data.List (foldl')
+
+type Arg name uni fun a = (Term name uni fun a, a)
+
+-- | Strip off arguments
+splitApplication :: Term name uni fun a -> (Term name uni fun a, [Arg name uni fun a])
+splitApplication = go []
+  where
+    go acc = \case
+      Apply ann fun arg -> go ((arg, ann) : acc) fun
```

Comment:
I know it's horrible nitpicking, but it's counter-intuitive to me that in constructor `arg` comes after `ann`, but in the tuple it's the other way around.